Support type-dependent search result highlighting via CSS (#12474)
sphinx-doc/sphinx: doc/development/html_themes/index.rst

   ``sphinx.html_themes`` entry_points feature.

Styling with CSS
----------------

> The **stylesheets** setting allows to add custom CSS files to a theme.

Reviewer: ```suggestion
.. tip::

  The :confval:`stylesheets` setting can be used to add custom CSS files to a theme.
```

Author: I've chosen to use the text but not put it in an admonition. First, the sentence describes the foundation of styling; I feel tips should be useful additional but not essential pieces of information. Second, I've turned the following paragraph into a `.. caution`. Having a section only consisting of two admonitions feels weird. There should be some "normal" information.

The structure of the HTML elements and their classes are currently not a
well-defined public API. Please infer them from inspecting the built HTML
pages. While we cannot guarantee full stability, they tend to be fairly

> stable.

Reviewer: I'd either suggestion a `caution` [admonition](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#admonitions-messages-and-warnings) here, or that we be brave and build consensus that we're OK with the classnames we've selected for this.

Reviewer: Nope, sorry - I'd misunderstood this as being specific to search results, when it clearly is not.  That would lead me towards the admonition suggestion.

> The structure of the HTML elements and their classes are currently not a
> well-defined public API. Please infer them from inspecting the built HTML
> pages. While we cannot guarantee full stability, they tend to be fairly
> stable.

Author: ```suggestion
.. caution::

   The structure of the HTML elements and their classes are currently not a
   well-defined public API. Please infer them from inspecting the built HTML
   pages. While we cannot guarantee full stability, they tend to be fairly
   stable.
```

Styling search result entries by category
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. versionadded:: 8.0

The search result items have classes indicating the context in which the
search term was found. You can use the CSS selectors:

> - ``ul.search li.context-index``:
>   If the search term is found in an index such as the glossary
> - ``ul.search li.context-object``:
>   If the search term a code object such as a python function
> - ``ul.search li.context-text``:
>   If the search term is found in plain documentation text
> - ``ul.search li.context-title``:
>   If the search term is found in a heading

Reviewer: ```suggestion
- ``ul.search li.context-index``:
  For results in an index, such as the glossary
- ``ul.search li.context-object``:
  For results in source code, like Python function definitions
- ``ul.search li.context-title``:
  For results found in section headings
- ``ul.search li.context-text``:
  For results found anywhere else in the documentation text
```

> The basic theme is intentional minimal and does not use them for styling.

Reviewer: ```suggestion
As a base for inheritance by other themes, the ``basic`` theme is intentionally minimal and does not define CSS rules using these.
```

(maybe too waffly, but I felt the additional background could be useful, despite the fact that we mention derived themes next-sentence)

Derived themes are encouraged to use these selectors as they see fit.

> For example, the following code would add context-dependent icons to the
> search result list:

Reviewer: ```suggestion
For example, the following stylesheet adds contextual icons to the
search result list:
```

   ``sphinx.html_themes`` entry_points feature.

Styling with CSS
----------------

> The **stylesheets** setting allows to add custom CSS files to a theme.

Author: ```suggestion
The :confval:`stylesheets` setting can be used to add custom CSS files to a theme.
```

Reviewer: My mistake, sorry - it seems that this isn't a traditional config setting; it's a theme configuration setting, and those appear to be different.  I'll try to figure out whether it _could_ be made reference-able.. but perhaps that should be reverted.
